instrumentation/github.com/Shopify/sarama/message.go

```diff
@@ -0,0 +1,94 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sarama
+
+import (
+	"github.com/Shopify/sarama"
+
+	"go.opentelemetry.io/otel/api/propagation"
+)
+
+var _ propagation.HTTPSupplier = (*ProducerMessageCarrier)(nil)
+var _ propagation.HTTPSupplier = (*ConsumerMessageCarrier)(nil)
+
+// ProducerMessageCarrier injects and extracts traces from a sarama.ProducerMessage.
+type ProducerMessageCarrier struct {
```

Comment:
Is there a need to treat the `ProducerMessageCarrier` and `ConsumerMessageCarrier` types differently?  They look to have the same implementation.  When I have added tracing to a system using Kafka messaging, using the Confluent Go library, I simply added the `Get()` and `Set()` methods to a type shadowing the library's message `Header` type that I used in both consumer and producer code.

Reply:
`;_;`